site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,666 @@
+            cy.visit(['sample', 'extensions']);
+
+            for (const test_case of test_info) {
+                const { g_id, g_title } = test_case;
+                const late_users = all_late_users[g_id];
+                late_day_exception[g_id] = {};
+
+                cy.get('#gradeable-select').select(g_title);
+
+                for (const user_id in late_users) {
+                    const days = Math.floor(Math.random() * 2) + 1;
+                    // update the number of late days
+                    cy.get('#user_id').type(user_id);
+                    cy.get('#late-days').type(days, {force: true});
+                    cy.get('#extensions-form')
+                        .find('a')
+                        .contains('Submit')
+                        .click();
+                    cy.wait(3000);
+                    late_day_exception[g_id][user_id] = days;
+                }
+            }
+        });
+
+        it('should make bulk late days has been emptied out', () => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('instructor');
+
+            for (const test_case of test_info) {
+                const { g_id, g_title } = test_case;
+                const late_users = all_late_users[g_id];
+
+                for (const user_id in late_users) {
+                    // Gradeable # of late days used should be empty
+                    cy.get(`[data-user="${user_id}"] > [data-before-content="${g_title}"]`)
+                        .then((cell) => expect(cell.text().trim()).to.equal(''));
+
+                    // Remaining late days isnt known, should be empty
+                    cy.get(`[data-user="${user_id}"] > [data-before-content="Late Days Remaining"]`)
+                        .then((cell) => expect(cell.text().trim()).to.equal(''));
+                }
+            }
+        });
+
+        it('should make sure late day status has updated', () => {
+            for (const test_case of test_info) {
+                const { g_id, g_title, days_allowed } = test_case;
+                const late_users = all_late_users[g_id];
+
+                for (const user_id in late_users) {
+                    cy.visit([]);
+                    cy.login(user_id);
+                    cy.visit(['sample', 'late_table']);
+
+                    // Wait for login change to take place
+                    cy.wait(3000);
+                    const status = predictedStatus(days_allowed, late_users[user_id] - late_day_exception[g_id][user_id], 0);
+
+                    // Find late day status within the row in the late day usage table
+                    cy.get('td[data-before-content="Event/Assignment"]')
+                        .contains(g_title)
+                        .siblings('td[data-before-content="Status"]')
+                        .contains(status)
+                        .should('exist');
+
+                    cy.logout();
+                }
+            }
+        });
```

Comment:
Instead of having this as a separate test, considering adding a JS function that has this code, and call it after the previous test (after line 322). 

(Similar comment to the use of this test code above)